rclcpp/include/rclcpp/executors/multi_threaded_executor.hpp

```diff
@@ -69,14 +69,26 @@ class MultiThreadedExecutor : public rclcpp::Executor
   void
   spin() override;
 
+  /**
+   * \sa rclcpp::Executor:spin() for more details
+   * \throws std::runtime_error when spin() called while already spinning
+   * @param exception_handler will be called for every exception in the processing threads
+   *
+   * The exception_handler can be called from multiple threads at the same time.
+   * The exception_handler shall rethrow the exception it if wants to terminate the program.
+   */
+  RCLCPP_PUBLIC
+  void
+  spin(std::function<void(const std::exception & e)> exception_handler);
```

Comment:
why this is dedicated to `MultiThreadedExecutor`? i think user application can find useful with other Executors as well.

Reply:
In a single thread executor, the user is the one that is creating the thread where we spin, so they can take care of handling the exception without any new API.
This is not the case for the threads spawned internally by an executor, such as in this case

Reply:
yeah i understand that, but user can use the same manner for any executor? i just though that would be useful to write the application.

Reply:
I don't care either way...

Reply:
@fujitatomoya @alsora can you come to an agreement, so that we can move forward with this PR ?

Reply:
I'm ok with adding it to all executors.

Reply:
Added spin API with exception handler to Executor class